src/Json.Schema.ToDotNet.UnitTests/DataModelGeneratorTests.cs

```diff
@@ -2363,7 +2369,152 @@ public int GetHashCode(C obj)
+  }
+}";
+            const string ExpectedClass =
+@"using System;
+using System.CodeDom.Compiler;
+using System.ComponentModel;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace N
+{
+    [DataContract]
+    [GeneratedCode(""Microsoft.Json.Schema.ToDotNet"", """ + VersionConstants.FileVersion + @""")]
+    public partial class C
+    {
+        /// <summary>
+        /// An integer property.
+        /// </summary>
+        [DataMember(Name = ""intProp"", IsRequired = false, EmitDefaultValue = false)]
+        public int IntProp { get; set; }
+
+        /// <summary>
+        /// An integer property with a default value.
+        /// </summary>
+        [DataMember(Name = ""intPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(42)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public int IntPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A number property.
+        /// </summary>
+        [DataMember(Name = ""numberProp"", IsRequired = false, EmitDefaultValue = false)]
+        public double NumberProp { get; set; }
+
+        /// <summary>
+        /// A number property with a default value.
+        /// </summary>
+        [DataMember(Name = ""numberPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(42.1)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public double NumberPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A string property.
+        /// </summary>
+        [DataMember(Name = ""stringProp"", IsRequired = false, EmitDefaultValue = false)]
+        public string StringProp { get; set; }
+
+        /// <summary>
+        /// A string property with a default value.
+        /// </summary>
+        [DataMember(Name = ""stringPropWithDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(""42"")]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string StringPropWithDefault { get; set; }
+
+        /// <summary>
+        /// A Boolean property.
+        /// </summary>
+        [DataMember(Name = ""boolProp"", IsRequired = false, EmitDefaultValue = false)]
+        public bool BoolProp { get; set; }
+
+        /// <summary>
+        /// A Boolean property with a true default value.
+        /// </summary>
+        [DataMember(Name = ""boolPropWithTrueDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(true)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public bool BoolPropWithTrueDefault { get; set; }
+
+        /// <summary>
+        /// A Boolean property with a false default value.
+        /// </summary>
+        [DataMember(Name = ""boolPropWithFalseDefault"", IsRequired = false, EmitDefaultValue = false)]
+        [DefaultValue(false)]
```

Comment:
this actually an entirely unnecessary attribution, I believe, as is the JsonProperty attribute that values. For cases where .NET init saves the day, EmitDefaultValue is sufficient. whether you want to handle this case is up to you. Note that there are .NET helpers for retrieving default values of object type, I believe, which might help you create conditional logic to avoid this code emit

Reply:
Nice idea. I filed [#63](https://github.com/Microsoft/jschema/issues/63), "Don't emit default-related attributes if the JSON schema default is the same as the .NET default", and will address it separately (but not with high priority -- it was important to get this far, for the sake of the SARIF schema changes, but we can certainly live without #63).

I found [this helper](https://stackoverflow.com/questions/2490244/default-value-of-a-type-at-runtime/2490274#2490274):
```
object GetDefaultValue(Type t)
{
    if (t.IsValueType)
        return Activator.CreateInstance(t);

    return null;
}
```



---
In reply to: [244508499](https://github.com/Microsoft/jschema/pull/62#discussion_r244508499) [](ancestors = 244508499)